lib/omnibus/software.rb

```diff
@@ -785,10 +785,24 @@ def with_standard_compiler_flags(env = {}, opts = {})
             "CFLAGS" => "-I#{install_dir}/embedded/include #{arch_flag} -O2 -fno-lto #{opt_flag}",
           }
         else
-          {
-            "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib",
+          flags = {
+            "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib -Wl,-rpath-link=#{install_dir}/embedded/lib",
             "CFLAGS" => "-I#{install_dir}/embedded/include -O2",
           }
+          if ENV["DD_CC"]
+            flags["CC"] = ENV["DD_CC"]
+            # Forward the toolchain env since some invoke tasks
+            # rely on it down the line
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ since this comment seems to apply to all the `if` statements, it should be moved to a more "global" place, or duplicated per.

Reply:
Good point, I took the opportunity to attempt another approach which will allow simpler additional variables if we need them